src/http.js

```diff
@@ -161,55 +162,145 @@ export function isFile(obj, navigatorObj) {
     }
     return false
   }
-  if (typeof File !== 'undefined') {
-    // eslint-disable-next-line no-undef
-    return obj instanceof File
+
+  if (typeof File !== 'undefined' && obj instanceof File) { // eslint-disable-line no-undef
+    return true
+  }
+  if (typeof Blob !== 'undefined' && obj instanceof Blob) { // eslint-disable-line no-undef
+    return true
   }
+  if (typeof Buffer !== 'undefined' && obj instanceof Buffer) {
+    return true
+  }
+
   return obj !== null && typeof obj === 'object' && typeof obj.pipe === 'function'
 }
 
-function formatValue(input, skipEncoding) {
```

Comment:
Since the key may change depending on the value, `formatValue` has been changed to a method that returns a key-value pair